tfrecord/writer.py

```diff
@@ -49,23 +57,34 @@ def masked_crc(data):
         return masked_bytes
 
     @staticmethod
-    def serialize_tf_example(datum):
-        """Serialize example into tfrecord.Example proto. 
+    def serialize_tf_example(datum: typing.Dict[str, typing.Tuple[typing.Any, str]]) -> bytes:
+        """Serialize example into tfrecord.Example proto.
+
+        Params:
+        -------
+        datum: dict
+            Dictionary of tuples of form (value, dtype). dtype can be
+            "byte", "float" or "int".
 
-        Args:
-            Datum is a dictionary of tuples of form (value, dtype). dtype can be "byte", "float" or "int".
         Returns:
+        --------
+        proto: bytes
             Serialized tfrecord.example to bytes.
         """
+        serialize = {
+            "byte": lambda f: example_pb2.Feature(
+                bytes_list=example_pb2.BytesList(value=f)),
+            "float": lambda f: example_pb2.Feature(
+                float_list=example_pb2.FloatList(value=f)),
+            "int": lambda f: example_pb2.Feature(
+                int64_list=example_pb2.Int64List(value=f))
+        }
+
         features = {}
         for key, (value, dtype) in datum.items():
-            feature = {
-                "byte": lambda f: example_pb2.Feature(bytes_list=example_pb2.BytesList(value=[f])),
-                "float": lambda f: example_pb2.Feature(float_list=example_pb2.FloatList(value=f)),
-                "int": lambda f: example_pb2.Feature(int64_list=example_pb2.Int64List(value=f))
-            }[dtype](value)
-            features[key] = feature
-
-        example_proto = example_pb2.Example(
-            features=example_pb2.Features(feature=features))
+            # Map type to list, otherwise they won't unpack
+            if not isinstance(value, (list, np.ndarray)):
```

Comment:
This is a breaking change. Let's not combine it with cleanup. It has to be made as a separate PR. Unless there's a use case for this I prefer to keep the old way.